tests/test_button.py

```diff
@@ -354,3 +357,89 @@ async def test_quirks_write_attr_buttons_uid(zha_gateway: Gateway) -> None:
     assert entity_btn_2.translation_key == "btn_2"
     assert entity_btn_2._unique_id_suffix == "btn_2"
     assert entity_btn_2._attribute_value == 2
+
+
+class OppleCluster(CustomCluster, ManufacturerSpecificCluster):
+    """Aqara manufacturer specific cluster."""
+
+    cluster_id = 0xFCC0
+    ep_attribute = "opple_cluster"
+
+    class ServerCommandDefs(zcl_f.BaseCommandDefs):
+        """Server command definitions."""
+
+        self_test: Final = zcl_f.ZCLCommandDef(
+            id=0x00, schema={"identify_time": t.uint16_t}
+        )
+
+
+async def test_cluster_handler_quirks_unnecessary_claiming(
+    zha_gateway: Gateway,
+) -> None:
+    """Test quirks button doesn't claim cluster handlers unnecessarily."""
+
+    registry = DeviceRegistry()
+    (
+        QuirkBuilder(
+            "Fake_Manufacturer_sensor_2", "Fake_Model_sensor_2", registry=registry
+        )
+        .replaces(OppleCluster)
+        .command_button(
+            FakeManufacturerCluster.ServerCommandDefs.self_test.name,
+            OppleCluster.cluster_id,
+            command_args=(5,),
+            translation_key="self_test",
+            fallback_name="Self test",
+        )
+        .add_to_registry()
+    )
+
+    zigpy_device = create_mock_zigpy_device(
```

Comment:
Another nitpick, feel free to ignore if it's too much trouble. Is there any way to test using JSON diagnostics from a real device? I'd like to phase out use of `create_mock_zigpy_device` and other synthetic testing setups in favor of using real devices.

Reply:
We can address this in the future IMO. The new/modified tests are all based on one very similar (existing) test. If so, we should change all of them then.

Maybe we can have some JSON diagnostics files in a separate folder that aren't real devices (and cannot be "modified" if we add new diagnostics). That way we could avoid `create_mock_zigpy_device`.

We could add a comment in the tests then to explain what the diagnostics/tests devices do.
I do think that manually modifying JSON files isn't as easy as using `create_mock_zigpy_device`. For these tests, `create_mock_zigpy_device` is kind of ideal.